pkg/log/handler.go

```diff
@@ -0,0 +1,45 @@
+package log
+
+import (
+	"fmt"
+	"github.com/applike/gosoline/pkg/cfg"
+	"time"
+)
+
+type Handler interface {
+	Channels() []string
+	Level() int
+	Log(timestamp time.Time, level int, msg string, args []interface{}, err error, data Data) error
+}
+
+type HandlerFactory func(config cfg.Config, handlerIndex int) (Handler, error)
+
+var handlerFactories = map[string]HandlerFactory{
+	"iowriter": handlerIoWriterFactory,
+}
+
+func AddHandlerFactory(name string, factory HandlerFactory) {
+	handlerFactories[name] = factory
+}
+
+func NewHandlersFromConfig(config cfg.Config) ([]Handler, error) {
+	settings := &LoggerSettings{}
+	config.UnmarshalKey("log", settings)
+
+	var ok bool
+	var err error
+	var handlerFactory HandlerFactory
+	var handlers = make([]Handler, len(settings.Handlers))
+
+	for i, handlerSettings := range settings.Handlers {
+		if handlerFactory, ok = handlerFactories[handlerSettings.Type]; !ok {
```

Comment:
Personally, I favour not to define the variables separately:
```
handlerFactory, ok := handlerFactories...
if !ok {
}```

Only saves a few lines, I know :D

Reply:
Personally, I like defining the variables separately cause I find it more readable :)